src/storage/CompositeStorage.cxx

```diff
@@ -118,7 +118,7 @@ CompositeStorage::Directory::Make(const char *uri)
 	Directory *directory = this;
 	while (*uri != 0) {
 		const std::string name = NextSegment(uri);
-		auto i = directory->children.emplace(std::move(name),
+		auto i = directory->children.emplace(name,
```

Comment:
Same here. The problem is the `const`, not the `std::move()`.

Reply:
Warning: std::move of the const variable 'name' has no effect; remove std::move() or make the variable non-const [hicpp-move-const-arg]

I guess I  can remove the const.

Reply:
Fixed.